examples/common/convert_c4.py

```diff
@@ -4,105 +4,359 @@
 """C4 streaming dataset conversion scripts."""
 import os
 import platform
-from argparse import ArgumentParser, Namespace
-from typing import Dict, Iterable
+from argparse import Action, ArgumentError, ArgumentParser, Namespace
+from enum import Enum
+from typing import Dict, Iterable, Optional
 
 import datasets as hf_datasets
+import numpy as np
 from streaming import MDSWriter
 from torch.utils.data import DataLoader, IterableDataset, get_worker_info
 from tqdm import tqdm
+from transformers import AutoTokenizer, PreTrainedTokenizerBase
+
+# Utilities and parsers
+
+class ConcatMode(Enum):
+    NO_CONCAT = 'NO_CONCAT'
+    CONCAT_TEXT = 'CONCAT_TEXT'
+    CONCAT_TOKENS = 'CONCAT_TOKENS'
+
+
+class EmptyOrNonexistentDirectory(Action):
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        prospective_dir = values
+        if os.path.isdir(prospective_dir) and len(
+                os.listdir(prospective_dir)) > 0:
+            raise ArgumentError(self,
+                                f'--out_root={prospective_dir} must be empty')
+        else:
+            setattr(namespace, 'out_root', prospective_dir)
 
 
 def parse_args() -> Namespace:
     """Parse commandline arguments."""
-    args = ArgumentParser()
-    args.add_argument('--out_root', type=str, required=True)
-    args.add_argument('--compression', type=str, default=None)
-    args.add_argument('--splits',
-                      nargs='+',
-                      default=['train', 'train_small', 'val'])
+    parser = ArgumentParser(description='Convert C4 into MDS format, optionally concatenating')
+    parser.add_argument('--out_root',
+                        type=str,
+                        required=True,
+                        action=EmptyOrNonexistentDirectory)
+    parser.add_argument('--compression', type=str, default=None)
+    parser.add_argument('--splits',
```

Comment:
multiple splits doesn't seem to be working (mentioned in dm)

Reply:
This should be working now, updating the example commands